### PR TITLE
Refactoring the method for displaying error and warning counts

### PR DESCRIPTION
**Pull request by ltp23**
In this small PR, I moved the method `get_error_and_warning_counts()` from `main.py` to inside the OutputManager class. This method is then called by the `simulate()` in the SimulationEngine class.

I also removed the method `show_error_warning_counts()` and kept the error count message to a minimum.

This is a follow-up PR to address additional comments from issue #1251 and the closed PR #1302

### CODE REVIEW

File: RUFAS/output_manager.py [hunk 1]
```diff
@@ -114,6 +114,7 @@ def __init__(self) -> None:
                     "function": self.__init__.__name__,
                 },
             )
+            self._stdout_queue: List[str] = []
```
Comment (ew3361zh) [inline RUFAS/output_manager.py]: Is this used anywhere?

**Review by ew3361zh — APPROVED** [review]
This looks good considering the feedback on #1302. I think the limited additional stdout is a good compromise for the time between now and when #1161 is addressed.

I wasn't sure the `self._stdout_queue` was needed as it's not referenced anywhere.

Nice work!

**Review by Isgotkowitz — APPROVED** [review]
LGTM, nice job Loi

File: RUFAS/output_manager.py (hunk 1 above)
Comment (ltp23) [reply RUFAS/output_manager.py]: I will remove that. I was working on managing the printout messages but then changed my mind and left it for another PR.

**Comment by ltp23** [comment]
I made another issue #1340 to handle the console messages in a more controlled manner.

**Pull request merged**